common/lib/xmodule/xmodule/video_module/video_utils.py

```diff
@@ -39,43 +41,41 @@ def create_youtube_string(module):
     ])
 
 
-# def get_video_from_cdn(cdn_base_url, original_video_url, cdn_branding_logo_url):
-# Not sure if this third variable is necessary...
-def get_video_from_cdn(cdn_base_url, original_video_url):
+def rewrite_video_url(cdn_base_url, original_video_url):
     """
-    Get video URL from CDN.
-
-    `original_video_url` is the existing video url.
-    Currently `cdn_base_url` equals 'http://api.xuetangx.com/edx/video?s3_url='
-    Example of CDN outcome:
-        {
-            "sources":
-                [
-                    "http://cm12.c110.play.bokecc.com/flvs/ca/QxcVl/u39EQbA0Ra-20.mp4",
-                    "http://bm1.42.play.bokecc.com/flvs/ca/QxcVl/u39EQbA0Ra-20.mp4"
-                ],
-            "s3_url": "http://s3.amazonaws.com/BESTech/CS169/download/CS169_v13_w5l2s3.mp4",
-        }
-    where `s3_url` is requested original video url and `sources` is the list of
-    alternative links.
+    Returns a re-written video URL for cases when an alternate source
+    has been configured and is selected using factors like
+    user location.
+
+    Re-write rules for country codes are specified via the
+    EDX_VIDEO_CDN_URLS configuration structure.
+
+    :param cdn_base_url: The scheme, hostname, port and any relevant path prefix for the alternate CDN,
+    for example: https://mirror.example.cn/edx
+    :param original_video_url: The canonical source for this video, for example:
+    https://cdn.example.com/edx-course-videos/VIDEO101/001.mp4
+    :return: The re-written URL
     """
 
-    if not cdn_base_url:
+    if (not cdn_base_url) or (not original_video_url):
         return None
 
-    request_url = cdn_base_url + urllib.quote(original_video_url)
+    parsed = urlparse(original_video_url)
+    # Contruction of the rewrite url is intentionally very flexible of input.
+    # For example, https://www.edx.org/ + /foo.html will be rewritten to
+    # https://www.edx.org/foo.html.
+    rewritten_url = cdn_base_url.rstrip("/") + "/" + parsed.path.lstrip("/")
+    validator = URLValidator()
 
     try:
-        cdn_response = requests.get(request_url, timeout=0.5)
```

Comment:
The timeout/404 was from the RPC call which no longer exists.
In the new flow, we rewrite URLs to them for the China case, they guarantee URL symmetry with us, if they cannot service a request they will fail fast and redirect to us.  We are eagerly pushing video updates to them to avoid the need to redirect.  Still, we'll need to monitor this.  We can get this data from the CDN.